client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { useState, useEffect } from 'react';
+import { Grid, CssBaseline, TextField, InputAdornment, Typography, Button } from '@material-ui/core';
+import SearchIcon from '@material-ui/icons/Search';
+import DateRangeIcon from '@material-ui/icons/DateRange';
+import ProfilePreview from './ProfilePreview/ProfilePreview';
+import generateMockData from './generateMockData';
+import { defaultUsers } from './mockData';
+import useStyles from './useStyles';
+
+const ProfileListings = (): JSX.Element => {
+  const classes = useStyles();
+  const [currentUsers, setCurrentUsers] = useState<number>(6);
+  const [mockData, setMockData] = useState<any>([]);
+
+  const defaultPreviews = defaultUsers.map((el, idx) => {
+    return (
+      <Grid item key={`default${idx}`}>
+        <ProfilePreview
+          img={el.img}
+          name={el.name}
+          subtitle={el.subtitle}
+          rating={el.rating}
+          description={el.description}
+          location={el.location}
+          payRate={el.payRate}
+        />
+      </Grid>
+    );
+  });
+
+  const renderedPreviews = mockData.map(
+    (data: { name: Record<string, unknown>; picture: Record<string, any>; id: Record<string, any> }[]) => {
+      return (
+        <Grid item key={data[0].id.value}>
+          <ProfilePreview
+            img={data[0].picture.large}
+            name={`${data[0].name.first} ${data[0].name.last}`}
+            subtitle={'Puppy power'}
+            rating={3}
+            description={'I like to take long walks on the beach with your dog.'}
+            location={'Ontario, Canada'}
+            payRate={15}
+          />
+        </Grid>
+      );
+    },
+  );
+
+  useEffect(() => {
+    generateMockData(currentUsers).then((res) => setMockData(res));
+  }, [currentUsers]);
+
+  return (
+    <Grid container spacing={1} component="main" className={classes.root}>
+      <CssBaseline />
```

Comment:
The `CssBaseline` is a global reset component and should be included only once, ideally in the App.tsx
If you are looking to implement it within the children, consider using `ScopedCssBaseline` instead

Ref: https://mui.com/api/scoped-css-baseline/